olp-cpp-sdk-core/src/cache/DiskCache.cpp

```diff
@@ -216,13 +216,23 @@ OpenResult DiskCache::Open(const std::string& data_path,
     status = leveldb::DB::Open(open_options, versioned_data_path, &db);
   }
 
-  // If the database is r/w and corrupted, attempt to repair & reopen
-  if ((status.IsCorruption() || status.IsIOError()) &&
-      RepairCache(versioned_data_path) == true) {
-    status = leveldb::DB::Open(open_options, versioned_data_path, &db);
-    if (status.ok()) {
-      database_.reset(db);
-      return OpenResult::Repaired;
+  if (status.IsCorruption()) {
```

Comment:
In the case of r/w database, which means mutable cache, we should not just abort. The mutable cache is an active cache and can be repaired.